src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -316,6 +316,17 @@ public Task<Stream> ReadAsStreamAsync(CancellationToken cancellationToken)
 
         protected abstract Task SerializeToStreamAsync(Stream stream, TransportContext? context);
 
+        protected virtual void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            // No choice but to do sync-over-async. Derived types should override this whenever possible.
+            // Thankfully most CreateContentReadStreamAsync implementations actually complete synchronously.
+            Stream source = CreateContentReadStreamAsync(cancellationToken).GetAwaiter().GetResult();
```

Comment:
Shouldn't there be a sync CreateContentReadStream API? It's default could fall back to sync-over-async if it needed to, but it sounds like many of them wouldn't.

Reply:
Resolved by throwing NSE as agreed in discussion bellow.